plugins/polio/api/vaccines/stock_management.py

```diff
@@ -78,11 +75,169 @@ class VaccineStockManagementReadWritePerm(GenericReadWritePerm):
+
+        results = []
+
+        for report in arrival_reports:
+            results.append(
+                {
+                    "date": report.arrival_report_date,
+                    "action": "PO #" + report.po_number if report.po_number else "Stock Arrival",
+                    "vials_in": report.vials_received,
+                    "doses_in": report.doses_received,
+                    "vials_out": None,
+                    "doses_out": None,
+                }
+            )
+
+        for report in destruction_reports:
+            results.append(
+                {
+                    "date": report.destruction_report_date,
+                    "action": f"{report.action} ({report.lot_number})"
+                    if len(report.action) > 0
+                    else f"Stock Destruction ({report.lot_number})",
+                    "vials_in": None,
+                    "doses_in": None,
+                    "vials_out": report.unusable_vials_destroyed,
+                    "doses_out": "N/A",
+                }
+            )
+
+        for report in incident_reports:
+            results.append(
+                {
+                    "date": report.date_of_incident_report,
+                    "action": report.stock_correction,
+                    "vials_in": None,
+                    "doses_in": None,
+                    "vials_out": report.unusable_vials + report.usable_vials,
+                    "doses_out": "N/A",
+                }
+            )
+
+        for movement in stock_movements:
+            if movement.usable_vials_used > 0:
+                results.append(
+                    {
+                        "date": movement.report_date,
+                        "action": "Form A - Vials Used",
+                        "vials_in": None,
+                        "doses_in": None,
+                        "vials_out": movement.usable_vials_used,
+                        "doses_out": "N/A",  # can't compute with current model
+                    }
+                )
+            elif movement.unusable_vials > 0:
+                results.append(
+                    {
+                        "date": movement.report_date,
+                        "action": "Form A - Unusable Vials",
+                        "vials_in": None,
+                        "doses_in": None,
+                        "vials_out": movement.unusable_vials,
+                        "doses_out": "N/A",  # can't compute with current model
+                    }
+                )
+            elif movement.missing_vials > 0:
+                results.append(
+                    {
+                        "date": movement.report_date,
+                        "action": "Form A - Missing Vials",
+                        "vials_in": None,
+                        "doses_in": None,
+                        "vials_out": movement.missing_vials,
+                        "doses_out": "N/A",  # can't compute with current model
```

Comment:
- `vials_in` is always `None`
- `doses_out` is always `None` or `N/A`.

Are those fields useful at all?

Reply:
Yes there will be cases in the other endpoint where vials_in is not None and about the doses crap I would need to fill it in but in the current state it's not "possible" .. Not sure how much we need it so I need feedback from client for that.